src/pybdy/profiler.py

```diff
@@ -299,10 +295,17 @@ def process_bdy(setup_filepath=0, mask_gui=False):
     if ln_tra:
         var_in["t"].extend(["votemper", "vosaline"])
 
+        if settings["interp"]:  # on-the-fly interp requires e3
+            var_in["t"].extend["e3t"]
+
     if ln_dyn2d or ln_dyn3d:
         var_in["u"].extend(["vozocrtx", "vomecrty"])
         var_in["v"].extend(["vozocrtx", "vomecrty"])
 
+        if settings["interp"]:  # on-the-fly interp requires e3
```

Comment:
I hadn't intended using the extract class to deal with grid data. Here, you're passing vertical scale factors as vector objects that will ultimately undergo a rotation! We. can discuss this later - but thought I'd make a note here in the mean time.

Reply:
Right, no problem. I was viewing .extend() as an "add variable" function. I'd not thought about concequences of rotation. The scale factors are often output as diagnostics in the same files as the velocities and we need to keep the e3u/e3v components with the velocities when supplying them as boundary conditions. It would be good if .extend() was more of a .add_variable() function, with an option to bypass rotation. That's its function for the grid_T files, right?

Reply:
you can view .extend() as an add variable 'function' ... it's just a list method to append variables that are then passed to the extract class. The vertical grid information should really be handled in the coords object, which is how it's set up in master. We haven't extended things to a 3D source depth yet (which will complicate things further) + there's no need to extract the scale factors over time as most of the time they are in some sort of time invariant grid/domain file. If you really want to account for a time variant coordinate then this can be achieved using the ssh.